src/commands/preview/mod.rs

```diff
@@ -43,9 +46,21 @@ pub fn preview(
     let script_id = upload(&mut target, user.as_ref(), sites_preview, verbose)?;
 
     let session = Uuid::new_v4().to_simple();
-    let preview_host = "example.com";
-    let https = true;
-    let https_str = if https { "https://" } else { "http://" };
+
+    let mut protocol = "https://";
+    let mut domain = "example.com";
+    let mut path = "";
+
+    let regex = Regex::new(r"((?:http|https)://)([^/\\]*)(.*)").unwrap();
+
+    if regex.is_match(url) {
+
+        let captures = regex.captures(url).unwrap();
+
+        protocol = captures.get(1).map_or("https://", |m| m.as_str());
+        domain = captures.get(2).map_or("example.com", |m| m.as_str());
+        path = captures.get(3).map_or("", |m| m.as_str());
```

Comment:
I think for this I'd recommend using the `Url` crate which will handle all of this parsing for you - https://docs.rs/url/2.1.1/url/struct.Url.html

You should be able to `let url = Url::parse(url);` and pass that to the preview function where you can then pull out `url.scheme()`, `url.domain()`, and `url.path()`

Reply:
Good thinking, I'll work on this now!

Reply:
Resolved this by using the `Url` crate as mentioned.

The `preview` command now passes a `Url` through to the handler.

A new struct has been created called `RequestPayload` which accepts a `Url` alongside the `method` and `body`. 

This struct now handles this functionality alongside dynamically generating the cookie based on the `script_id`